app/scripts/metamask-controller.js

```diff
@@ -5502,6 +5503,25 @@ export default class MetamaskController extends EventEmitter {
               },
             },
           ),
+        grantPermittedChainsPermission: (chainIds) => {
+          this.permissionController.grantPermissionsIncremental({
+            subject: {
+              origin,
+            },
+            requestData: {
+              approvedChainIds: chainIds,
+            },
+            approvedPermissions: {
+              [PermissionNames.permittedChains]: {
+                caveats: [
+                  CaveatFactories[CaveatTypes.restrictNetworkSwitching](
+                    chainIds,
+                  ),
+                ],
+              },
```

Comment:
Seems a bit strange that we have to specify the chain IDs twice 🤔 Seems like the permission specification strictly uses `requestData`